chainstate/src/detail/tokens/check_utils.rs

```diff
@@ -13,11 +13,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use common::chain::ChainConfig;
+use common::{
+    chain::{tokens::TokenIssuanceVersion, ChainConfig},
+    primitives::BlockHeight,
+};
 use tx_verifier::error::TokenIssuanceError;
 use utils::ensure;
 
 fn check_is_text_alphanumeric(str: &[u8]) -> bool {
+    match String::from_utf8(str.to_vec()) {
+        Ok(text) => text.chars().all(char::is_alphanumeric),
+        Err(_) => false,
+    }
+}
+
+fn check_is_text_ascii_alphanumeric(str: &[u8]) -> bool {
     match String::from_utf8(str.to_vec()) {
         Ok(text) => text.chars().all(|c| c.is_ascii_alphanumeric()),
         Err(_) => false,
```

Comment:
Two points here:
* Can use `Result::is_ok_and` but that has been stabilised fairly recently, not sure we want to require such a recent compiler version
* `is_ascii_alphanumeric` may be somewhat restrictive for description for exmaple. It would reject anything with a space or usual punctuation `.,;:!?` in it. I think ticker and name need a separate set of valid characters different from description.

Reply:
Btw there's `std::str::from_utf8` that lets you convert `&[u8]` to `&str` without allocating a `String` in a validating way.

Reply:
> is_ascii_alphanumeric may be somewhat restrictive for description for exmaple. 

that's by design to avoid consequences that we don't understand at the moment. Can easily be changed later

> Btw there's std::str::from_utf8

nice thanks

Reply:
also simplified `check_is_text_ascii_alphanumeric`

Reply:
> Can easily be changed later

Is it not a part of consensus rules?

Reply:
I mean with a fork